File: backend/src/main/scala/kpn/api/common/LocationChangeSetInfo.scala
package kpn.api.common

import kpn.api.common.changes.details.ChangeKey

case class LocationChangeSetInfo(
  rowIndex: Long,
  key: ChangeKey,
  comment: Option[String],
  happy: Boolean,
  investigate: Boolean,
  locationChanges: Seq[LocationChangesInfo]
)

File: backend/api/src/main/java/kpn/api/common/LocationChangeSetInfo.java
package kpn.api.common;

import kpn.api.common.LocationChangesInfo;
import kpn.api.common.changes.details.ChangeKey;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record LocationChangeSetInfo(
  Long rowIndex,
  ChangeKey key,
  Optional<String> comment,
  Boolean happy,
  Boolean investigate,
  ImmutableList<LocationChangesInfo> locationChanges
) {
}
